src/MaxText/input_pipeline/_hf_data_processing.py

```diff
@@ -47,6 +47,28 @@ def vision_sft_preprocessing_pipeline(
   if config.enable_data_shuffling:
     dataset = dataset.shuffle(seed=config.data_shuffle_seed)
 
+  # If multiple image columns are provided, merge them into a single 'images' column.
+  if isinstance(image_column, list):
+    # First, truncate the list of image columns if it exceeds the configured max.
+    if config.max_num_images_per_example > 0 and len(image_column) > config.max_num_images_per_example:
+      image_column = image_column[: config.max_num_images_per_example]
+
+    # Define a mapping function that accepts the list of columns as an argument.
+    def merge_image_columns(example, column_names):
```

Comment:
Could we make it a general merge_columns function and put in _input_pipeline_utils.py?